lua/neotest-gradle/hooks/test_configuration_resolver.lua

```diff
@@ -0,0 +1,43 @@
+local logger = require("neotest.logging")
+
+-- Function: extract_test_dir
+-- Description: Extracts the directory name located between 'src' and 'java' in a given file path.
+-- Parameters:
+--   path (string): The full file path.
+--   project_dir (string): The project directory path.
+-- Returns:
+--   test_dir (string) or nil: The name of the test directory or nil if not found.
```

Comment:
Could you please use [EmmyLua annotations](https://emmylua.github.io/annotation.html) as the rest of the code base? 🙂 
This gives nice editor support by language servers (type checks, signatures, better completion, ...). Plus it contributes to a more homogeneous code base.

Additionally, please try to avoid abbreviations in code. Even such commonly known ones like `dir` when you mean `directory`. Could be also `direction` or `derivation`. You automatically need less comments and annotation.

Like:
```suggestion
--- Extracts the directory name located between `src` and `java` in a given file path.
---
--- @param test_file_path string
--- @param project_directory string
--- @return string | nil name of the test directory or `nil` if not found
```

Reply:
ok, I am not lua developer so any comment is nice to have